src/test/kotlin/org/jetBrains/research/tasktracker/tracking/TaskFileHandlerTest.kt

```diff
@@ -0,0 +1,163 @@
+        }
+        assert(projectToTaskToFiles[project]?.let { it.size == 2 } ?: false)
+        assertNoThrowable {
+            taskFileHandler.disposeTask(project, task1)
+        }
+        assert(projectToTaskToFiles[project]?.let { it.size == 1 } ?: false)
+        assert(projectToTaskToFiles[project]?.let { it.keys.first() == task2 } ?: false)
+        assertNoThrowable {
+            taskFileHandler.disposeTask(project, task2)
+        }
+        assert(projectToTaskToFiles.isEmpty())
+    }
+
+    fun testAddVirtualFileListener() {
+        val file1 = File("${project.basePath}/test1")
+        val file2 = File("${project.basePath}/test2")
+        val directory = File("${project.basePath}/tests/")
+        ApplicationManager.getApplication().runWriteAction {
+            FileUtil.createIfDoesntExist(file1)
+            FileUtil.createIfDoesntExist(file2)
+            FileUtil.createDirectory(directory)
+        }
+        val virtualFile1 = LocalFileSystem.getInstance().findFileByIoFile(file1) ?: error("file must exist")
+        val virtualFile2 = LocalFileSystem.getInstance().findFileByIoFile(file2) ?: error("file must exist")
+        val virtualDirectory =
+            LocalFileSystem.getInstance().findFileByIoFile(directory) ?: error("file must exist")
+        assert(virtualFile1.exists())
+        assert(virtualFile2.exists())
+        assert(virtualDirectory.exists())
+        val method = getPrivateFunc("addVirtualFileListener")
+        assertNoThrowable {
+            method.call(taskFileHandler, virtualFile1)
+            method.call(taskFileHandler, virtualFile2)
+            method.call(taskFileHandler, virtualDirectory)
+        }
+    }
+
+    fun testRemoveVirtualFileListener() {
+        val file = File("${project.basePath}/test")
+        val directory = File("${project.basePath}/tests/")
+        ApplicationManager.getApplication().runWriteAction {
+            FileUtil.createIfDoesntExist(file)
+            FileUtil.createDirectory(directory)
+        }
+        val virtualFile = LocalFileSystem.getInstance().findFileByIoFile(file) ?: error("file must exist")
+        val virtualDirectory =
+            LocalFileSystem.getInstance().findFileByIoFile(directory) ?: error("file must exist")
+        assert(virtualFile.exists())
+        assert(virtualDirectory.exists())
+        val methodRemove = getPrivateFunc("removeVirtualFileListener")
+        assertFailsWith<Throwable> {
+            methodRemove.call(taskFileHandler, virtualFile)
+        }
+        val methodAdd = getPrivateFunc("addVirtualFileListener")
+        assertNoThrowable {
+            methodAdd.call(taskFileHandler, virtualFile)
+            methodRemove.call(taskFileHandler, virtualFile)
+            methodRemove.call(taskFileHandler, virtualDirectory)
+        }
+        assertFailsWith<Throwable> {
+            methodRemove.call(taskFileHandler, virtualFile)
+        }
+    }
+
+    fun testGetOrCreateFile() {
```

Comment:
You can also add asserts inside tests that I described above to avoid using reflection here